### ISSUE TITLE
🐛 Bug Report: SonarQube plugin does not compile

### ISSUE THREAD

**Issue by drodil** [backstage/backstage#16358]
### 📜 Description

Updating to v1.11.0 breaks compilation in plugin-sonarqube-react

### 👍 Expected behavior

tsc compiles correctly

### 👎 Actual Behavior with Screenshots

yarn run v1.22.19
$ tsc --skipLibCheck false --incremental false
node_modules/@backstage/plugin-sonarqube/dist/index.d.ts:13:10 - error TS2305: Module '"@backstage/plugin-sonarqube-react"' has no exported member 'FindingSummary'.

13 import { FindingSummary } from '@backstage/plugin-sonarqube-react';
            ~~~~~~~~~~~~~~

node_modules/@backstage/plugin-sonarqube/dist/index.d.ts:16:10 - error TS2305: Module '"@backstage/plugin-sonarqube-react"' has no exported member 'SonarQubeApi'.

16 import { SonarQubeApi } from '@backstage/plugin-sonarqube-react';
            ~~~~~~~~~~~~

Found 2 errors in the same file, starting at: node_modules/@backstage/plugin-sonarqube/dist/index.d.ts:13

error Command failed with exit code 1.


### 👟 Reproduction steps

Update to backstage v1.11.0
Run `tsc --skipLibCheck false --incremental false`

### 📃 Provide the context for the Bug.

_No response_

### 🖥️ Your Environment

OS:   Darwin 22.1.0 - darwin/arm64
node: v16.19.0
yarn: 1.22.19
cli:  0.22.2 (installed)
backstage:  1.11.0

Dependencies:
  @backstage/app-defaults                                   1.2.0
  @backstage/backend-app-api                                0.4.0
  @backstage/backend-common                                 0.17.0, 0.18.2
  @backstage/backend-dev-utils                              0.1.0
  @backstage/backend-plugin-api                             0.4.0
  @backstage/backend-tasks                                  0.4.3
  @backstage/backend-test-utils                             0.1.34
  @backstage/catalog-client                                 1.3.1
  @backstage/catalog-model                                  1.2.0
  @backstage/cli-common                                     0.1.11
  @backstage/cli                                            0.22.2
  @backstage/config-loader                                  1.1.8
  @backstage/config                                         1.0.6
  @backstage/core-app-api                                   1.5.0
  @backstage/core-components                                0.11.2, 0.12.4
  @backstage/core-plugin-api                                1.4.0
  @backstage/dev-utils                                      1.0.12
  @backstage/errors                                         1.1.4
  @backstage/eslint-plugin                                  0.1.0
  @backstage/integration-aws-node                           0.1.1
  @backstage/integration-react                              1.1.10
  @backstage/integration                                    1.4.2
  @backstage/plugin-api-docs                                0.9.0
  @backstage/plugin-app-backend                             0.3.42
  @backstage/plugin-auth-backend                            0.18.0
  @backstage/plugin-auth-node                               0.2.11
  @backstage/plugin-catalog-backend-module-bitbucket-server 0.1.7
  @backstage/plugin-catalog-backend-module-msgraph          0.5.0
  @backstage/plugin-catalog-backend                         1.7.2
  @backstage/plugin-catalog-common                          1.0.11
  @backstage/plugin-catalog-graph                           0.2.27
  @backstage/plugin-catalog-import                          0.9.5
  @backstage/plugin-catalog-node                            1.3.3
  @backstage/plugin-catalog-react                           1.3.0
  @backstage/plugin-catalog                                 1.8.0
  @backstage/plugin-github-actions                          0.5.15
  @backstage/plugin-jenkins-backend                         0.1.32
  @backstage/plugin-jenkins-common                          0.1.13
  @backstage/plugin-jenkins                                 0.7.14
  @backstage/plugin-org                                     0.6.5
  @backstage/plugin-permission-common                       0.7.3
  @backstage/plugin-permission-node                         0.7.5
  @backstage/plugin-permission-react                        0.4.10
  @backstage/plugin-proxy-backend                           0.2.36
  @backstage/plugin-scaffolder-backend                      1.11.0
  @backstage/plugin-scaffolder-common                       1.2.5
  @backstage/plugin-scaffolder-node                         0.1.0
  @backstage/plugin-scaffolder-react                        1.1.0
  @backstage/plugin-scaffolder                              1.11.0
  @backstage/plugin-search-backend-module-elasticsearch     1.1.3
  @backstage/plugin-search-backend-module-pg                0.5.3
  @backstage/plugin-search-backend-node                     1.1.3
  @backstage/plugin-search-backend                          1.2.3
  @backstage/plugin-search-common                           1.2.1
  @backstage/plugin-search-react                            1.5.0
  @backstage/plugin-search                                  1.1.0
  @backstage/plugin-sonarqube-react                         0.1.2
  @backstage/plugin-sonarqube                               0.6.3
  @backstage/plugin-techdocs-backend                        1.5.3
  @backstage/plugin-techdocs-module-addons-contrib          1.0.10
  @backstage/plugin-techdocs-node                           1.5.0
  @backstage/plugin-techdocs-react                          1.1.3
  @backstage/plugin-techdocs                                1.5.0
  @backstage/plugin-user-settings                           0.7.0
  @backstage/release-manifests                              0.0.8
  @backstage/test-utils                                     1.2.5
  @backstage/theme                                          0.2.17
  @backstage/types                                          1.0.2
  @backstage/version-bridge                                 1.0.3


### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by drodil** [comment]
What's going on with the plugin? Checking the node_modules/@backstage/plugin-sonarqube-react/dist/index.d.ts:
```ts
/* Excluded from this release type: FindingSummary */

/** @public */
export declare const isSonarQubeAvailable: (entity: Entity) => boolean;

/* Excluded from this release type: MetricKey */

/* Excluded from this release type: Metrics */

/** @public */
export declare const SONARQUBE_PROJECT_KEY_ANNOTATION = "sonarqube.org/project-key";

/* Excluded from this release type: SonarQubeApi */

/* Excluded from this release type: sonarQubeApiRef */

/* Excluded from this release type: SonarUrlProcessorFunc */

/* Excluded from this release type: useProjectInfo */

export { }
```

**Comment by benjdlambert** [comment]
@drodil looks like these are labeled as `@alpha`.

https://github.com/backstage/backstage/blob/0870577bd4ac6922ff999a0207c2aed71512d499/plugins/sonarqube-react/src/hooks/useProjectInfo.ts#L22-L36

So they would need to be imported like this:

```ts
import { useProjectInfo, MetricKey } from '@backstage/plugin-sonarqube-react/alpha`
```

I think that this is a problem here as we don't have #15816 yet, and these imports are not being re-written in the main codebase.

@Rugvip is this an oversight or a bug, should this have worked?

**Comment by pjungermann** [comment]
@benjdlambert it worked before 1.11.0

**Comment by benjdlambert** [comment]
Yep it's basically this commit: https://github.com/backstage/backstage/commit/6310eacc11 which is because it's using `/alpha` imports without actually using the `/alpha` path because it's not supported in the monorepo yet.

**Comment by benjdlambert** [comment]
You should be able to roll-back the package.json bump for `@backstage/plugin-sonarqube` at least to get you unblocked from the bump.

**Comment by benjdlambert** [comment]
OK, so to clarify a little bit, this is the issue what's happening here: https://github.com/backstage/backstage/pull/10128

It's currently not possible to re-export `/alpha` types in the current monorepo setup, and we have no linting or anything around this, just e2e tests, which unfortunately don't catch it for this plugin as it's not installed in the `create-app` bundle, so there's an action for us to maybe add better linting around this.

We have some options here, we either move some imports around and move them from `/alpha` and make sure that we're not re-exporting any `/alpha` types. Maybe it doesn't make sense to even have the `/alpha` imports for a package that isn't 1.x anways?

Or we revert the original PR, and create a new release, so that upgrading to 1.11 is easier as you don't have to lock the sonarqube dependency.

We haven't had time to test the https://github.com/backstage/backstage/pull/15816 `subpath exports` properly yet, so I think rushing that through is probably the last thing we want to do.

**Comment by pjungermann** [comment]
downgrading to `@backstage/plugin-sonarqube` 0.6.2 will resolve the issue for now (even still keeping `@backstage/plugin-sonarqube-react` 0.1.2)

**Comment by benjdlambert** [comment]
Chatted with the other @backstage/maintainers, and I think we're going to opt for removing the `/alpha` types from the package for now. It's really tricky to fix right now until we have the `subpath exports` which we don't want to rush through as it's quite a lot of tooling changes and we want to test to make sure this works as expected first. I will create a PR to move these types now, and get the fix out today.

**Comment by benjdlambert** [comment]
Closing as got released part of the bonus 1.11.1 release.
